chainer/reporter.py

```diff
@@ -0,0 +1,325 @@
+import collections
+import contextlib
+
+import numpy
+import six
+
+from chainer import cuda
+from chainer import variable
+
+
+class Reporter(object):
+
+    """Object to which observed values are reported.
+
```

Comment:
We may need the definition of `reporter`, `observation` and `observer` and relationship between them at first.